repository/src/main/java/org/apache/atlas/repository/store/graph/v2/EntityGraphMapper.java

```diff
@@ -2216,6 +2216,11 @@ private void addInternalProductAttr(AttributeMutationContext ctx, List<Object> c
         AtlasVertex toVertex = ctx.getReferringVertex();
         String toVertexType = getTypeName(toVertex);
 
+        if(currentElements.isEmpty() && createdElements.isEmpty() && deletedElements.isEmpty()){
```

Comment:
There are couple of cases where we pass currentElements null explicitly, add null check as well for currentElements